### PR TITLE
Document GUI Standards.

### PR DESCRIPTION
**Pull request by ghost**
## Description of Changes:

Document GUI standards with some examples. 

## To Test:

```bash
cd /path/to/my/local/garnet/repo/
git fetch origin pull/<PULL_REQUEST_NUMBER>/head:pr<PULL_REQUEST_NUMBER>
git switch pr<PULL_REQUEST_NUMBER>
conda activate <garnet_environment>
python -m pytest <item_to_test>
```

## Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

---

## Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

### CODE REVIEW

File: docs/source/GUI_standards.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+In Qt applications, a best practice is to disable QPushButton, and similar objects,
+when associated required fields are empty or fail validation. This approach prevents
+users from attempting form submission with incomplete or incorrect data, promoting a
+more user-friendly experience by guiding them to address validation issues before proceeding.
+It ensures data integrity and contributes to a smoother interaction flow within the application.
+
+Buttons should also be given meaningful names, such as “Load Experiment Plan” rather than
+generic names like “Load”. If multiple buttons exist in the View, it should be clear to the
+use which input/output fields correspond to each button. This can be achieved through grouping
```
Comment (mpatrou) [inline docs/source/GUI_standards.rst:84]: user

**Comment by hetrickjm** [comment]
I recommend renaming the section called **Algorithm Input Validation** to something more like **Data Input Validation**.  I like that there are two types of validation

1. Field validation which focuses on making sure what the user entered into the field is correct for the filed type. and
2. Validating the data itself.

It is not clear that all data entered by a user will be directly used in an algorithm.  I would prefer a more generic term for this type of validation and not focus (limit) it to algorithms.

Error Handling, perhaps better would be visual error handling.  This is not the error handling strategy, but more how to inform the user when something goes wrong.  We should add a link in here to the Error Handling strategy (I think this is being covered by loggging).

File: src/garnet/base/base_lineedit.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+"""BaseLineEdit
+------------
```
Comment (mpatrou) [inline src/garnet/base/base_lineedit.py]: please create a folder called utility and inside create another called UIelements or similar; include the base_lineedit.py and base_statusbar.py there.

**Review by mpatrou — APPROVED** [review]
works as expected

**Pull request merged**